maestro-client/src/main/java/maestro/Maestro.kt

```diff
@@ -393,21 +396,7 @@ class Maestro(private val driver: Driver) : AutoCloseable {
     }
 
     fun waitForAppToSettle(initialHierarchy: ViewHierarchy? = null): ViewHierarchy {
-        var latestHierarchy = initialHierarchy ?: viewHierarchy()
-        repeat(10) {
-            val hierarchyAfter = viewHierarchy()
-            if (latestHierarchy == hierarchyAfter) {
-                val isLoading = latestHierarchy.root.attributes.getOrDefault("is-loading", "false").toBoolean()
-                if (!isLoading) {
-                    return hierarchyAfter
-                }
-            }
-            latestHierarchy = hierarchyAfter
-
-            MaestroTimer.sleep(MaestroTimer.Reason.WAIT_TO_SETTLE, 200)
-        }
-
-        return latestHierarchy
+        return if(isIOS) screenshotWaitForAppToSettle(initialHierarchy) else genericWaitForAppToSettle(initialHierarchy)
```

Comment:
Can we ask the driver to do waitForAppToSettle using polymorphism? The method calls and method names has become confusing 🤔